score-e2e/src/test/java/org/oagi/score/e2e/TS_29_BIEUplifting/TC_29_1_BIEUplifting.java

```diff
@@ -0,0 +1,2693 @@
+        });
+    }
+
+    private void thisAccountWillBeDeletedAfterTests(AppUserObject appUser) {
+        this.randomAccounts.add(appUser);
+    }
+
+    @Test
+    public void TAs() {
+        test_TA_29_1_1();
+        test_TA_29_1_2_BIE_Uplift();
+        test_TA_29_1_3();
+        test_TA_29_1_4_and_TA_29_1_5a_and_TA_29_1_6a();
+        test_TA_29_1_5c_and_TA_29_1_7_and_TA_29_1_8();
+        test_TA_29_1_5d_and_TA_29_1_6ab();
+        test_TA_29_1_9a();
+        test_TA_29_1_9b_and_TA_29_1_9c();
+        test_TA_29_1_10a();
+        test_TA_29_1_10b();
+        test_TA_29_1_11a_and_TA_29_11b();
+        test_TA_29_1_12();
+    }
+
+    @Test
+    public void test_TA_29_1_1() {
+        usera = getAPIFactory().getAppUserAPI().createRandomEndUserAccount(false);
+        thisAccountWillBeDeletedAfterTests(usera);
+        HomePage homePage = loginPage().signIn(usera.getLoginId(), usera.getPassword());
+        BIEMenu bieMenu = homePage.getBIEMenu();
+        UpliftBIEPage upliftBIEPage = bieMenu.openUpliftBIESubMenu();
+        upliftBIEPage.setSourceBranch(curr_release);
+        assertThrows(TimeoutException.class, () -> upliftBIEPage.setTargetBranch(prev_release));
+    }
+
+    @Test
+    public void test_TA_29_1_2_BIE_Uplift() {
+        developer = getAPIFactory().getAppUserAPI().createRandomDeveloperAccount(false);
+        thisAccountWillBeDeletedAfterTests(developer);
+        usera = getAPIFactory().getAppUserAPI().createRandomEndUserAccount(false);
+        thisAccountWillBeDeletedAfterTests(usera);
+        TopLevelASBIEPObject BIEUserbProduction = null;
+        if (testingBIEs.containsKey("BIEUserbProduction")) {
+            BIEUserbProduction = testingBIEs.get("BIEUserbProduction");
+        } else {
+            preconditions_TA_29_1_2_Uplift_BIEUserbProduction();
+            BIEUserbProduction = testingBIEs.get("BIEUserbProduction");
+        }
+        HomePage homePage = loginPage().signIn(developer.getLoginId(), developer.getPassword());
+        BIEMenu bieMenu = homePage.getBIEMenu();
+        UpliftBIEPage upliftBIEPage = bieMenu.openUpliftBIESubMenu();
+        // Uplift Production BIE
+        upliftBIEPage.openPage();
+        upliftBIEPage = bieMenu.openUpliftBIESubMenu();
+        upliftBIEPage.setSourceBranch(prev_release);
+        upliftBIEPage.setTargetBranch(curr_release);
+        upliftBIEPage.setState("Production");
+        upliftBIEPage.setPropertyTerm(BIEUserbProduction.getPropertyTerm());
+        upliftBIEPage.hitSearchButton();
+        WebElement tr = upliftBIEPage.getTableRecordAtIndex(1);
+        WebElement td = upliftBIEPage.getColumnByName(tr, "select");
+        click(td);
+        click(upliftBIEPage.getNextButton());
+        WebDriverWait wait = new WebDriverWait(getDriver(), Duration.ofSeconds(180));
+        wait.until(ExpectedConditions.invisibilityOfElementLocated(By.xpath("//*[contains(@class, 'loading-container')]")));
```

Comment:
Use `invisibilityOfLoadingContainerElement` in PageHelper. See https://github.com/OAGi/Score/blob/163e8822328386355b3b82009273bc0408070029/score-e2e/src/main/java/org/oagi/score/e2e/impl/page/module/CreateModuleSetReleasePageImpl.java#L108